Photon-Tinker/ScanBarcodeViewController.m

```diff
@@ -100,15 +104,32 @@ - (void)viewDidLoad
 
     [self.view bringSubviewToFront:_highlightView];
     [self.view bringSubviewToFront:_label];
-    [self.view bringSubviewToFront:circleView];
-    [self.view bringSubviewToFront:overlayButton];
-    [self.view bringSubviewToFront:overlayImageView];
-    [self.view bringSubviewToFront:cancelButton];
+    [self.view bringSubviewToFront:_circleView];
+    [self.view bringSubviewToFront:_overlayButton];
+    [self.view bringSubviewToFront:_overlayImageView];
+    [self.view bringSubviewToFront:_cancelButton];
 
     
 }
 
 
+- (void)viewWillLayoutSubviews {
+    [super viewWillLayoutSubviews];
+
+    if (@available(iOS 11.0, *)) {
```

Comment:
Will it look ok on <iOS 11 devices?

Reply:
Yes, this only modifies layout for devices having safeArea (which is only available in ios 11). Original layout code is still being run for older iOS versions.